### ISSUE TITLE
bump LSP4J  version  0.14.0 to 0.20.1

### ISSUE THREAD

**Issue by w6et** [eclipse/lemminx#1497]
LSP4J [0.20.1](https://projects.eclipse.org/projects/technology.lsp4j/releases/0.20.1) has now been released
lemminx now just use 0.14.0，i suggest bump the version  0.14.0 to 0.20.1
**benefit：**
- some bugfix from 0.14.0 to 0.20.1
- lsp4j  runtime base on java8(compatible)

**code change:**
org.eclipse.lemminx.XMLTextDocumentService#prepareRename
```
Either<Range, PrepareRenameResult> either = getXMLLanguageService().prepareRename(xmlDocument, params.getPosition(), cancelChecker);
if (either != null) {
	if(either.isLeft()) {
		return Either3.forFirst((Range)either.get());
	}else {
		return Either3.forSecond((PrepareRenameResult)either.get());
	}
}else {
	return null;
}
//return getXMLLanguageService().prepareRename3(xmlDocument, params.getPosition(), cancelChecker);
```
#在 LSP4J 0.15.0
Return type of textDocument/prepareRename changed from Either<Range, PrepareRenameResult>
to Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>

**Comment by angelozerr** [comment]
@w6et is there any chance that you create a PR?

**Comment by w6et** [comment]
> @w6et is there any chance that you create a PR?

[/eclipse/lemminx/pull/1498](https://github.com/eclipse/lemminx/pull/1498)

**Comment by datho7561** [comment]
Closed in #1498
